### DISCUSSION TITLE
Support for Nuphy and/or https://usevia.app/

### DISCUSSION THREAD

**Discussion by discentem** [manna-harbour/miryoku#333] [Q&A] [answered]
- Would it be possible to add support for Nuphy Keyboards? In particular the Nuphy Air75 v2. 
- With regard to usevia, I did see your previous comment here about json files not being useful for multiple keyboard models: https://github.com/manna-harbour/miryoku/discussions/74#discussioncomment-2256020 and I see the open issue to track via support: https://github.com/manna-harbour/miryoku/issues/90. If it is indeed possible to do today somehow, would you be able to add instructions somewhere in the repo for configuring even without json files?

**Comment by manna-harbour** [comment, marked as answer]
To use a keyboard with Miryoku QMK, you first need to be able to build the default keymap for that keyboard with QMK. Nuphy keyboards don't seem to be in QMK, but if you might be able to find a fork that has support, e.g. https://github.com/qmk/qmk_firmware/pull/22751.

After that you need a mapping in QMK. For row staggered keyboards, only support for `60_ansi` is included by default https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#60_ansi. For other row-staggered layouts you'd need to add the equivalent mapping. Alternatively, you could use Miryoku KMonad https://github.com/manna-harbour/miryoku_kmonad.

For Via / Vial see https://github.com/manna-harbour/miryoku/discussions/238#discussioncomment-9504852. If there's any change to the status, https://github.com/manna-harbour/miryoku/issues/90 will be updated, so subscribe to that issue for notifications.
